From 2c83ace48ffe1bb6b5ef0b6590bb2394ccda5687 Mon Sep 17 00:00:00 2001
From: Justin Ding <justin.ding@netgear.com>
Date: Tue, 3 Jul 2018 22:48:29 -0700
Subject: [PATCH] network: disable and enable ipv6 when switching from auto to
 manual.

IPV6 dhcp IP still exits if changing from auto to manual, try to disable
and then enable IPV6 to clear IPV6 dhcp IP.
---
 src/network.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/network.c b/src/network.c
index 56fe24f..e4038d6 100644
--- a/src/network.c
+++ b/src/network.c
@@ -1876,7 +1876,8 @@ int __connman_network_clear_ipconfig(struct connman_network *network,
 		break;
 	case CONNMAN_IPCONFIG_METHOD_AUTO:
 		release_dhcpv6(network);
-		if (type == CONNMAN_IPCONFIG_TYPE_IPV6)
+		__connman_ipconfig_disable_ipv6(ipconfig);
+		__connman_ipconfig_enable_ipv6(ipconfig);
 			break;
 		/* fall through */
 	case CONNMAN_IPCONFIG_METHOD_DHCP:
-- 
2.1.4

